### PR TITLE
Return Expression HTML elements

### PR DESCRIPTION
**Pull request by njseeto**
[Trello card](https://trello.com/c/HBC7iR7g/1729-backend-api-to-return-expression-html-elements)

### Return API Expression HTML elements 
Returns the HTML snippets for the operator and the answers for a particular expression.
We pass the Page object through so the Expression class has access to it.
Error handling has been included, a 422 is produced when the endpoint is hit incorrectly.

Co-authored-by: Brendan Butler <brendan.butler@digital.justice.gov.uk>

### Use MetadataPresenter gem 2.1.1

### CODE REVIEW

File: app/views/branches/_expression_answers.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      id: "branch_conditionals_attributes_#{conditional_index}_expressions_attributes_#{expression_index}_operator"
+    }  %>
+  <%= f.select :answers, @expression.answers,
+    { include_blank: t('branches.select_answer') },
+    {
+      class: 'govuk-select',
+      name: "branch[conditionals_attributes][#{conditional_index}][expressions_attributes][#{expression_index}][answer]",
+      id: "branch_conditionals_attributes_#{conditional_index}_expressions_attributes_#{expression_index}_answer"
+    } %>
+</div>
```
Comment (tomas-stefano) [inline app/views/branches/_expression_answers.html.erb:16]: I am not sure if we can use `child_index` or not. 🤔 
Even if we can't, can we dry the naming moving into a helper method? Wdyt?

Comment (njseeto) [reply app/views/branches/_expression_answers.html.erb:16]: sounds good - can use a helper method to dry it up

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -114,3 +114,9 @@ Style/SlicingWithRange:
 # SupportedStyles: use_perl_names, use_english_names
 Style/SpecialGlobalVars:
   EnforcedStyle: use_perl_names
+
+# Offense count: 1
+# Cop supports --auto-correct.
+Style/WordArray:
+  Exclude:
+    - 'app/models/expression.rb'
```
Comment (njseeto) [inline .rubocop_todo.yml:122]: I'm not 100% sure this is the best way to disable rubocop on the one line, open to any ideas

Comment (brenetic) [reply .rubocop_todo.yml:122]: `# rubocop:disable SomeRule` if you're doing it inline

Comment (brenetic) [reply .rubocop_todo.yml:122]: it is done in the file which has the offending code, not the rubocop config

Comment (njseeto) [reply .rubocop_todo.yml:122]: oh i see, that is much simpler, will change that now

File: app/controllers/api/expressions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      @page_with_component ||= service.page_with_component(params[:component_id])
+    end
+
+    def component
+      page_with_component.find_component_by_uuid(params[:component_id])
+    end
+
+    delegate :type, to: :component, prefix: true
+    helper_method :component_type
+    def expression_name(type)
```
Comment (brenetic) [inline app/controllers/api/expressions_controller.rb:32]: new line above please

File: app/controllers/api/expressions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      "branch[conditionals_attributes][#{params[:conditional_index]}]" \
+      "[expressions_attributes][#{params[:expression_index]}][#{type}]"
+    end
+    helper_method :expression_name
+
+    def expression_id(type)
```
Comment (brenetic) [inline app/controllers/api/expressions_controller.rb:38]: will `type` call the delegated type above? not sure

Comment (njseeto) [reply app/controllers/api/expressions_controller.rb:38]: i don't think so, it renders correctly in the DOM.
But I can see how the naming can be confusing - perhaps I can change the argument from `type` to (** grabs the thesaurus **) `attribute`?

**Pull request merged**